ra2ce/analysis/damages/damage_functions/damage_fraction_uniform.py

```diff
@@ -71,23 +75,28 @@ def from_csv(self, path: Path, sep=",") -> None:
 
 
         """
-        self.name = path.stem
-        self.raw_data = pd.read_csv(path, index_col=0, sep=sep)
-        self.origin_path = path  # to track the original path from which the object was constructed; maybe also date?
+        _name = path.stem
+        _raw_data = pd.read_csv(path, index_col=0, sep=sep)
+        _origin_path = path  # to track the original path from which the object was constructed; maybe also date?
 
         # identify unit and drop from data
-        self.hazard_unit = self.raw_data.index[0]
-        self.data = self.raw_data.drop(
-            self.hazard_unit
+        _hazard_unit = _raw_data.index[0]
+        _data = _raw_data.drop(
+            _hazard_unit
         )  # Todo: This could also be a series instead of DataFrame
 
         # convert data to floats
-        self.data = self.data.astype("float")
-        self.data.index = self.data.index.astype("float")
+        _data = _data.astype("float")
+        _data.index = _data.index.astype("float")
 
-        self.convert_hazard_severity_unit()
+        _damage_fraction = cls(
+            name=_name, hazard_unit=_hazard_unit, data=_data, origin_path=_origin_path
+        )
+        _damage_fraction._convert_hazard_severity_unit()
```

Comment:
is this always needed? If so then it needs to become a `__post_init__` operation, more [here](https://docs.python.org/3/library/dataclasses.html#dataclasses.__post_init__)

Reply:
Nice suggestion